app/models/experimental_subject.coffee

```diff
@@ -0,0 +1,192 @@
+                        Experiments.sources[subjectID]=Experiments.SOURCE_INSERTED
+      .fail =>
+        AnalyticsLogger.logError "500", "Couldn't load next experimental subjects", "error"
+    else
+      backgroundFetcher = new $.Deferred
+      backgroundFetcher.resolve null
+    backgroundFetcher.promise()
+
+  # mark subject as seen in experiment server, and remove any "used" subjects from our queue.
+  @markAsSeen: (subjectID) ->
+    userID = User.current?.zooniverse_id
+    seenNotifier = new $.Deferred
+    try
+      if Experiments.sources[subjectID]==Experiments.SOURCE_INSERTED
+        url = Experiments.EXPERIMENT_SERVER_URL + 'experiment/' + Experiments.ACTIVE_EXPERIMENT + '/participant/' + userID + '/insertion/' + subjectID
+      else if Experiments.sources[subjectID]==Experiments.SOURCE_RANDOM
+        url = Experiments.EXPERIMENT_SERVER_URL + 'experiment/' + Experiments.ACTIVE_EXPERIMENT + '/participant/' + userID + '/random'
+      else
+        AnalyticsLogger.logError "409", "Couldn't POST subject "+subjectID+" to mark as seen", "error"
+        return null
+      $.post(url)
+      .then (participant) =>
+        Experiments.currentParticipant = participant
+        # ensure any previously seen subjects by this participant are removed from the queue if present.
+        for seenID in Experiments.currentParticipant.insertion_subjects_seen
+          for queuedSubject of ExperimentalSubject.all
+            if seenID==queuedSubject.zooniverseId
+              ExperimentalSubject.where(zooniverseId: seenID).delete
+        seenNotifier.resolve participant
+      .fail =>
+        AnalyticsLogger.logError "500", "Couldn't POST subject "+subjectID+" to mark as seen", "error"
+        seenNotifier.resolve null
+    catch error
+      AnalyticsLogger.logError "500", "Couldn't POST subject "+subjectID+" to mark as seen", "error"
+      seenNotifier.resolve null
+    seenNotifier.promise()
+
+  # get the next subject IDs for the specified user ID from the experiment server	(assumes "interesting" cohort)
+  @getNextSubjectIDs: (numberOfSubjects) ->
+    userID = User.current?.zooniverse_id
+    subjectIDsFetcher = new $.Deferred
+    try
+      $.get(Experiments.EXPERIMENT_SERVER_URL + 'experiment/' + Experiments.ACTIVE_EXPERIMENT + '/participant/' + userID + '/next/' + numberOfSubjects)
+      .then (data) =>
+        subjectIDsFetcher.resolve data
+      .fail =>
+        AnalyticsLogger.logError "500", "Couldn't retrieve next subjects", "error"
+        subjectIDsFetcher.resolve null
+    catch error
+      AnalyticsLogger.logError "500", "Couldn't GET next subjects", "error"
+      subjectIDsFetcher.resolve null
+    subjectIDsFetcher.promise()
+
+  # for interesting cohort users, we will show a selection of known interesting and random unknown subjects.
+  @next: (callback) =>
+    if Experiments.ACTIVE_EXPERIMENT?
+      if Experiments.ACTIVE_EXPERIMENT == "SerengetiInterestingAnimalsExperiment1"
```

Comment:
I'd combind these two into ifs into one statement. So you can have one else clause. 

Indeed testing if the ACTIVE_EXPERIMENT is non-null should be unnecessary since if it is null it'll just fail to equal the experiment name